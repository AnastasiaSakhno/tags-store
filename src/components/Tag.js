import React from 'react'
import PropTypes from 'prop-types'

const Tag = ({ linkId, name }) => {
  return (<span className="tag">{ name }</span>)
}

Tag.propTypes = {
  linkId: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired
}

export default Tag
