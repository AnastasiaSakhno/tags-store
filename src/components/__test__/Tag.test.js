import React from 'react'
import renderer from 'react-test-renderer'
import Tag from '../Tag'
import uniqId from '../../utils/uniq-id'

const linkId = uniqId()
const tag = {
  linkId: linkId,
  name: 'tag'
}
const onDestroy = () => {}

it('renders correctly', () => {
  const tree = renderer
    .create(<Tag name={ tag.name } linkId= { linkId } key={ linkId } onDestroy={ onDestroy } />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
