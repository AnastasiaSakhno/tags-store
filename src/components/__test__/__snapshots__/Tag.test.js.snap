// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<span
  className="tag"
>
  tag
  <span
    className="iconbutton-box"
    onClick={[Function]}
  >
    <i
      className="fa fa-remove "
    />
  </span>
</span>
`;
